src/GloryDuckTables2019Lkl.cc

```diff
@@ -0,0 +1,403 @@
+  Margin(level); cout << "                 <sv>_min = " << fsvMin[0]        << " [cm^3 s^-1]" << endl;
+  Margin(level); cout << "                 <sv>_max = " << fsvMax[fNMasses-1] << " [cm^3 s^-1]" << endl;
+}
+
+///////////////////////////////////////////////////////////////////////////////////////////////////////
+// 
+// Plot a canvas with all the -2logL, ie an array
+// of graphs (1 per mass) with -2logL vs <sv>
+// 
+// Return a pointer to the canvas, in case you want to modify it
+//
+TCanvas* GloryDuckTables2019Lkl::PlotInputData() 
+{
+  const Int_t nlines = 5;  // number of lines in parabolas canvas
+  Int_t ncols = TMath::Ceil(fNMasses/Float_t(nlines)); // number of columns in parabolas canvas
+
+  // create and divide canvas
+  TCanvas* canvas = new TCanvas("inputDataCanvas","-2logL value per energy bin and energy flux",ncols*250,nlines*250);
+  canvas->Divide(ncols,nlines);
+
+  for(Int_t ibin=0;ibin<fNMasses;ibin++)
+    {
+      ParabolaLkl* prbla   = (ParabolaLkl*)GetSample(ibin);
+      TGraph*      grprbla = prbla->GetParabola(kFALSE);
+
+      canvas->cd(ibin+1);
+      TH1I *dummypr = new TH1I(Form("dummypr_%d",ibin),Form("-2logLkl vs <sv> for m_{DM}=%.1f \\ and <sv>=[%.1f,%.1f] [cm^3 s^-1]",fMass[ibin],fsvMin[ibin],fsvMax[ibin]),1,grprbla->GetX()[1]*0.1,grprbla->GetX()[grprbla->GetN()-1]);
+      dummypr->SetDirectory(0);
+      dummypr->SetStats(0);
+      dummypr->SetXTitle("<sv> [cm^3 s^-1]");
+      dummypr->SetYTitle("#Delta(-2logL)");
+      dummypr->SetMinimum(0);
+      dummypr->SetMaximum(10);
+      dummypr->DrawCopy();
+      gPad->SetLogx();
+      delete dummypr;
+
+      // plot -2logLkl vs <sv> 
+      grprbla->Draw("l");
+      gPad->SetGrid();
+      gPad->Modified();
+      gPad->Update();
+    }
+  return canvas;
+}
+
+////////////////////////////////////////////////////////////////////////
+//
+// Set mass index to the current active one
+//
+void GloryDuckTables2019Lkl::SetActiveMass(Double_t mass)
+{
+  for (Int_t counter = 0; counter < fNMasses; counter++)
+    {
+      if(TMath::Abs(fMass[counter]-mass) < 1e-3)
+        {
+          fActiveMass = counter;
+          return;
+        }
+    }
+
+  cout << "GloryDuckTables2019Lkl::SetActiveMass() Message: Mass " << mass << " doesn't exist." << endl;
+  return;
+}
+
+////////////////////////////////////////////////////////////////////////
+//
+// Likelihood function (-2logL) for one mass
+// To be minimized by TMinuit
+// Free parameters:
+// par[0] = g
+//
+void gloryDuckTables2019Lkl(Int_t &fpar, Double_t *gin, Double_t &f, Double_t *par, Int_t iflag)
+{
+  // to avoid warnings only
+  gin[0]*=1;
+  fpar*=1;
+  iflag*=1;
+  f = 0;
+
+  // get internal object
+  GloryDuckTables2019Lkl* mylkl = dynamic_cast<GloryDuckTables2019Lkl*>(minuit->GetObjectFit());
+
+  // g is the global scale for dPhidESignal
+  Double_t g = par[0];
+
+  // get likelihood for current active mass
+  Lkl* sample = mylkl->GetSample(mylkl->GetActiveMass());
```

Comment:
This won't work, because sample will be interpreted as a Lkl object, not as ParabolaLkl one, use the dynamic cast as in the FermiTables2016Lkl case: 
ParabolaLkl* sample = dynamic_cast<ParabolaLkl*>mylkl->GetSample(mylkl->GetActiveMass());